kymatio/scattering1d/conv_primitive.py

```diff
@@ -0,0 +1,27 @@
+import torch.nn as nn
+import torch.nn.functional as F
+
+class Conv1dFFTPrimative(nn.Module):
+    def __init__(self, backend):
+        super(Conv1dFFTPrimative, self).__init__()
+        self.name = 'torch'
+        self.backend = backend
+
+    def preprocess_signal(self, x, pad_left=None, pad_right=None):
```

Comment:
I would go with `padding` and then a tuple of paddings or a list of tuples of paddings.

The reason I say this is that this fourier primitive can then eventually become N-D and can be used for all dimensionalities. Or we can inherit the different dimensionalities from it and have the inherited class fix the dimension

Reply:
You are smart.